lib/src/diktat/java/com/diffplug/spotless/glue/diktat/DiktatFormatterFunc.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021-2022 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.glue.diktat;
+
+import java.io.File;
+import java.util.*;
+
+import org.cqfn.diktat.ruleset.rules.DiktatRuleSetProvider;
+
+import com.pinterest.ktlint.core.KtLint;
+import com.pinterest.ktlint.core.KtLint.Params;
+import com.pinterest.ktlint.core.LintError;
+import com.pinterest.ktlint.core.RuleSet;
+
+import com.diffplug.spotless.FormatterFunc;
+
+import kotlin.Unit;
+import kotlin.jvm.functions.Function2;
+
+public class DiktatFormatterFunc implements FormatterFunc.NeedsFile {
+
+	private final List<RuleSet> rulesets;
+	private final Map<String, String> userData;
+	private final Function2<? super LintError, ? super Boolean, Unit> formatterCallback;
+	private final boolean isScript;
+
+	private final ArrayList<LintError> errors = new ArrayList<>();
+
+	public DiktatFormatterFunc(boolean isScript, Map<String, String> userData) {
+		rulesets = Collections.singletonList(new DiktatRuleSetProvider().get());
+		this.userData = userData;
+		this.formatterCallback = new FormatterCallback(errors);
+		this.isScript = isScript;
+	}
+
+	static class FormatterCallback implements Function2<LintError, Boolean, Unit> {
+		private final ArrayList<LintError> errors;
+
+		FormatterCallback(ArrayList<LintError> errors) {
+			this.errors = errors;
+		}
+
+		@Override
+		public Unit invoke(LintError lintError, Boolean corrected) {
+			if (!corrected) {
+				errors.add(lintError);
+			}
+			return null;
+		}
+	}
+
+	@Override
+	public String applyWithFile(String unix, File file) throws Exception {
+		errors.clear();
+		userData.put("file_path", file.getAbsolutePath());
+		String result = KtLint.INSTANCE.format(new Params(
+				file.getName(),
```

Comment:
as per #1189, shouldn't this be `file.getAbsolutePath()`? Or are you intentionally breaking up the `compileOnly` thing from the "fix the bug" thing?

Reply:
Yes, I didn't want to mix two changes in a single PR. But if that's acceptable and since both changes are pretty minor, I can push that fix here too.